### PR TITLE
US4B - Indoor Directions

### LINKED ISSUE #20: US4B: As a user, I want to see the shortest path from start to end point.
Multiple pathways are unnecessary

### LINKED ISSUE #73: US4B - Implement module with algorithm logic
Implement the algorithm which will calculate the shortest path into a React module. Should make algorithm work for one floor as proof of concept. Associated with user story US4B.

### LINKED ISSUE #74: US4B - Leverage simplest path module to work for all floors
Using the module implemented in issue #73, ensure that the shortest path algorithm function for indoor direction for each floor of every campus building.

### LINKED ISSUE #75: US4B - Transitions between indoor/outdoor mode
Implement a module which determines the closest entrance of a building when in outdoor mode, and which determines the closest exit of a building when in indoor mode.

### PR DESCRIPTION
**Pull request by Belize22**
Unit tests to be completed later this evening.
Supports single floor directions and multi-floor directions. Keep in mind that anything related to disability directions is related to US4E and room input is related to US4C.

### CODE REVIEW

**Comment by eufekt** [comment]
Vanier library second floor does not load. Loads on US4D which was merged to this PR branch.

File: assets/svgReactNative/missing_details/Ve1.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import * as React from 'react';
-import Svg, { G, Path } from 'react-native-svg';
+import * as React from "react";
```
Comment (eufekt) [inline assets/svgReactNative/missing_details/Ve1.js]: This happens all the time, can we decide on single or double quote? My linter says single all the time

File: components/map/building/buildingWithFloors/dijkstraPathfinder.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+   * shortest distance.
+   */
+  nodeDistance(startNodeId, endNodeId, adjacencyGraph) {
+    const deltaXSquared = (adjacencyGraph[endNodeId].x - adjacencyGraph[startNodeId].x) ** 2;
+    const deltaYSquared = (adjacencyGraph[endNodeId].y - adjacencyGraph[startNodeId].y) ** 2;
+    return Math.sqrt(deltaXSquared + deltaYSquared);
+  }
+};
+
+export default dijkstraPathfinder;
```
Comment (eufekt) [inline components/map/building/buildingWithFloors/dijkstraPathfinder.js:148]: this is art

**Review by eufekt — APPROVED** [review]
This proof of concept is working

File: assets/svgReactNative/missing_details/Ve1.js (hunk 1 above)
Comment (gisellemartel) [reply assets/svgReactNative/missing_details/Ve1.js]: Should be single quote according to the linter setting for this project

**Comment by trevornag** [comment]
I think it's a good proof of concept, we will work more on it in sprint 4

**Review by trevornag — APPROVED** [review]

**Pull request merged**